c_src/erl_optics.c

```diff
@@ -46,14 +48,50 @@ static int load(ErlNifEnv *env, void **priv_data, ERL_NIF_TERM load_info) {
     atom_dist = enif_make_atom(env, "dist");
     atom_histo = enif_make_atom(env, "histo");
     atom_quantile = enif_make_atom(env, "quantile");
-    poller = optics_poller_alloc();
-    ctx = enif_alloc(sizeof(struct map_ctx));
-    if (poller) {
-      optics_poller_backend(poller, (void *) ctx, backend_eo, backend_free_eo);
-    }
+    carbon_poller = false;
+    erlang_poller = false;
     return 0;
 }
 
+static ERL_NIF_TERM eo_register_carbon_poller(
+    ErlNifEnv *env, int argc, const ERL_NIF_TERM argv[])
+{
+    if (!poller){
+        poller = optics_poller_alloc();
```

Comment:
The on-demand creation of the poller object still bothers me a bit and I feel like it might be a tad bit less error prone to just always create the object in the `load` function instead of trying to create it on demand. Note that allocating the object does very little except call `malloc` and `hostname` so even if we end up never using the poller, the cost isn't huge.

Note that I still can't properly formulate why this bothers me so feel free to ignore me. I do think it has to do with the fact that the interface is not quite what I would like it to be but I believe we've compromised here in order to get shit done.

Reply:
The problem I have with creating the poller in the load function is that I couldn't find a way to make it explicit in case of failure (load returns an int, not an erlang term).

Reply:
Ah yes, I didn't think about that. Yep, this is annoying. Nevermind then.